solutions/src/main/java/ru/sberbank/school/task08/KryoManager.java

```diff
@@ -26,27 +27,38 @@ public KryoManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        Kryo kryo = new Kryo();
+        kryo = new Kryo();
         KryoSerializer kryoSerializer = new KryoSerializer();
         kryo.register(MapState.class, kryoSerializer);
+        kryo.register(InstantiatableEntity.Type.class);
+        kryo.register(InstantiatableEntity.Status.class);
+        kryo.register(GameObject.class);
+        kryo.register(ArrayList.class);
     }
 
     @Override
-    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
-
-        try (FileOutputStream fos = new FileOutputStream(filename);
+    public void saveGame(@NonNull String filename, @NonNull MapState<GameObject> gameState) throws SaveGameException {
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
              Output out = new Output(fos)) {
             kryo.writeObject(out, gameState);
+        } catch (NullPointerException | FileNotFoundException ex) {
+            throw new SaveGameException("File not found");
+        } catch (KryoException e) {
+            throw new SaveGameException("Kryo ex");
         } catch (IOException ex) {
             ex.printStackTrace();
         }
     }
 
     @Override
-    public MapState<GameObject> loadGame(String filename) throws SaveGameException {
-        try (FileInputStream fis = new FileInputStream(filename);
+    public MapState<GameObject> loadGame(@NonNull String filename) throws SaveGameException {
+        try (FileInputStream fis = new FileInputStream(filesDirectory + File.separator + filename);
              Input in = new Input(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a486380032>*